src/practices/Java/JavaLog4JConfiguredPractice.ts

```diff
@@ -0,0 +1,67 @@
+import { IPractice } from '../IPractice';
+import { ErrorFactory } from '../../lib/errors/ErrorFactory';
+import { PracticeEvaluationResult, PracticeImpact, ProgrammingLanguage } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import * as xml2js from 'xml2js';
+import * as properties from 'properties';
+import yaml from 'js-yaml';
+
+@DxPractice({
+  id: 'Java.Log4JConfigured',
+  name: 'Use Java Logger Configuration Files',
+  impact: PracticeImpact.small,
+  suggestion: 'Configure your Log4J using an xml, json, yaml, properties files; or use in-code configuration',
+  reportOnlyOnce: true,
+  url: 'https://logging.apache.org/log4j/2.x/manual/configuration.html',
+  dependsOn: { practicing: ['Java.LoggerUsedPractice'] },
+})
+export class JavaLog4JConfiguredPractice implements IPractice {
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.Java || ctx.projectComponent.language === ProgrammingLanguage.Kotlin;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (!ctx.fileInspector) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const dotExtensions = new RegExp('.(xml|json|yaml|yml|properties)', 'i');
+    const configurationFiles = await ctx.fileInspector.scanFor(dotExtensions, '/', { shallow: false });
+    const configurationKeys = ['configuration', 'log4j:configuration', 'rootLogger', 'logger', 'loggers', 'properties', 'appender'];
+    if (configurationFiles) {
+      for (const file of configurationFiles) {
+        if (file.baseName === 'log4j' || file.baseName === 'log4j2') {
+          const fileContents = await ctx.fileInspector.readFile(file.path);
+          let parsedContents = {};
+          try {
+            switch (file.extension) {
+              case '.xml':
+                parsedContents = await xml2js.parseStringPromise(fileContents);
+                break;
+              case '.json':
+                parsedContents = await JSON.parse(fileContents);
+                break;
+              case '.yaml':
```

Comment:
What about case `.yml` ?

Reply:
👍 @ryzzaki 🙏

Reply:
@prokopsimek @adelkahomolova sorry about that, it's fixed using a case fall-through and I added one more test to cover it :) I also reformated the internal error message.